<launch>
	<arg name="pkg_name" value="gd_interface"/>
	<arg name="node_name" default="gd_standerled_ni"/>
	<arg name="veh" doc="Name of vehicle. ex: megaman"/>
	<arg name="local" default="false" doc="true for running everything you possibly can on laptop. false for everything running on vehicle."/>
    <arg name="config" default="baseline"/>
    <arg name="param_file_name" default="default"/>
    <arg name="classes" default="7"/>


    <include file="$(find duckietown)/machines"/>
    <group ns="$(arg veh)">      
	    <remap from="gd_standerled_ni/led_classes" to="gd_prediction2led/led_classes"/> 
        <!--<remap from="gd_prediction2led/caffe_predictions" to="gd_L2R_text_text_node/caffe_predictions"/> -->
        <node unless="$(arg local)" machine="$(arg veh)" name="$(arg node_name)" pkg="$(arg pkg_name)" type="$(arg node_name).py" output="screen" clear_params="true" required="true">
            <rosparam command="load" file="$(find guidedog)/config/$(arg config)/$(arg pkg_name)/$(arg node_name)/$(arg param_file_name).yaml"/>
        </node>
        <node if="$(arg local)" name="$(arg node_name)" pkg="$(arg pkg_name)" type="$(arg node_name).py" output="screen" clear_params="true" required="true">
		    <rosparam command="load" file="$(find guidedog)/config/$(arg config)/$(arg pkg_name)/$(arg node_name)/$(arg param_file_name).yaml"/>
        </node>
    </group>
	

</launch>
